src/pyk/kcfg/kcfg.py

```diff
@@ -968,6 +969,24 @@ def prune(self, node_id: str) -> None:
         for node in nodes:
             self.remove_node(node.id)
 
+    def shortest_path_between(self, source_node_id: str, target_node_id: str) -> tuple[Successor, ...] | None:
+        paths = self.paths_between(source_node_id, target_node_id)
+        return sorted(paths, key=(lambda path: len(path)))[0]
+
+    def path_constraints(self, final_node_id: str) -> KInner:
+        path = self.shortest_path_between(self.get_unique_init().id, final_node_id)
+        if path is None:
+            raise ValueError(f'No path found to specified node: {final_node_id}')
+        curr_constraint: KInner = mlTop()
+        for edge in reversed(path):
+            if type(edge) is KCFG.Split:
+                assert len(edge.targets) == 1
+                _, csubst = edge.targets[0]
+                curr_constraint = mlAnd([csubst.subst.ml_pred, csubst.constraint, curr_constraint])
+            if type(edge) is KCFG.Cover:
+                curr_constraint = mlAnd([edge.csubst.constraint, edge.csubst.subst.apply(curr_constraint)])
+        return curr_constraint
```

Comment:
```suggestion
        return mlAnd(flatten_label('#And', curr_constraint))
```

This will get rid of the annoying `#Top #And ...` at the beginning of the constraint in your example.

Reply:
It looks like `flatten_label` doesn't remove the `#Top` term.

Reply:
I  changed it to use `ml_pred_to_bool` and then `simplify_bool` to simplify the output. I'm not sure if we have a preference of format for the output, but this might make it more readable.